### PR TITLE
Feat/logs to kafka

### PR DESCRIPTION
**Pull request by lepsalex**
* configurable kafka log appending via `kafka` profile which uses `log4j2-kafka.xml`
  * there are four env variables, readme has been updated to document this
  * running the app normally and/or for tests wil just output the logs to console, running with the `kafka` profile will send them to kafka
* all logging has been refactored to go through the GraphLogger class and its various static methods wherever possible
* transaction management has also been refactored to use `GraphTransitAuthority` methods that deal with the transaction but also remove it from the registry

### CODE REVIEW

**Review by jaserud — APPROVED** [review]
LGTM

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -6,7 +6,11 @@ rabbit:
       port: 5672
       managementPort:
 
-logging.level.root: INFO
+logging:
+  level:
+    root: INFO
+    org.icgc_argo.workflowgraphnode: INFO
```
Comment (andricDu) [inline src/main/resources/application.yml:12]: :+1:

File: src/main/resources/log4j2-kafka.xml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            <AppenderRef ref="kafkaDebugInfoLogAppender"/>
+            <AppenderRef ref="kafkaWarningErrorLogAppender"/>
+            <!--stdout/stderr included for testing log4j edits and having console output during development-->
+            <AppenderRef ref="stdout"/>
+            <AppenderRef ref="stderr"/>
+        </Root>
+        <!-- Do not let org.apache.kafka log to a Kafka appender on DEBUG level. That will cause recursive logging -->
+        <Logger name="org.apache.kafka" level="INFO"/>
+    </Loggers>
+</Configuration>
```
Comment (andricDu) [inline src/main/resources/log4j2-kafka.xml:53]: NEWLINE :fire:

File: src/main/resources/log4j2.xml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+            <ThresholdFilter level="WARN" onMatch="ACCEPT" onMismatch="DENY"/>
+        </Console>
+    </Appenders>
+    <Loggers>
+        <Root level="INFO">
+            <AppenderRef ref="stdout"/>
+            <AppenderRef ref="stderr"/>
+        </Root>
+    </Loggers>
+</Configuration>
```
Comment (andricDu) [inline src/main/resources/log4j2.xml:20]: :fire:

**Review by andricDu — APPROVED** [review]
Good stuff. Just the newlines.

**Pull request merged**